Gems/Prefab/PrefabDependencyViewer/Code/Tests/PrefabDependencyTreeGenerationTests.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PrefabDependencyViewerEditorSystemComponent.h>
+#include <PrefabDependencyTree.h>
+
+#include <DirectedGraph.h>
+#include <PrefabDependencyViewerFixture.h>
+
+#include <AzCore/std/algorithm.h>
+
+namespace PrefabDependencyViewer
+{
+    using DirectedGraph = Utils::DirectedGraph;
+    using TestComponent = PrefabDependencyViewerEditorSystemComponent;
+    using Outcome       = AZ::Outcome<PrefabDependencyTree, const char*>;
+    using NodeList      = AZStd::vector<Utils::Node*>;
+
+    bool str_equal(const char* expected, const char* result)
+    {
+        return (AZStd::string)expected == (AZStd::string)result;
+    }
+
+    void EXPECT_STR_EQ(const char* expected, const char* result)
```

Comment:
Google Test has a build in string check `EXPECT_STREQ`, why not use that?

Reply:
I wasn't aware of it's existence. Now I'm using it but since the MetaData class is returning a string_view that method I just had to make a few changes to this method in order to prevent a duplication and refactoring.

```
    void EXPECT_STR_EQ(const char* expected, AZStd::string_view result)
    {
        EXPECT_STREQ(expected, result.data());
    }

```

Reply:
Note that `AZStd::string_view` doesn't guarantee null-terminated strings so you can't just use `.data()` on it's own, the length has to be taken into account as well.

Reply:
Resolved